akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/RequestingRecoveryPermit.scala

```diff
@@ -34,12 +33,15 @@ private[akka] class RequestingRecoveryPermit[C, E, S](override val setup: Behavi
     with JournalInteractions[C, E, S]
     with SnapshotInteractions[C, E, S] {
 
-  onRequestingRecoveryPermit(setup.context)
-
   def createBehavior(): Behavior[InternalProtocol] = {
+    val instCtx =
+      setup.instrumentation.beforeRequestRecoveryPermit(setup.context.self)
+
     // request a permit, as only once we obtain one we can start replaying
     requestRecoveryPermit()
 
+    setup.instrumentation.afterRequestRecoveryPermit(setup.context.self, instCtx)
```

Comment:
@pvlugter We have `aroundCreateBehavior` which is calling those two beforeRequestRecoveryPermit and afterRequestRecoveryPermit. Can I collapse them in the SPI to only one method, or is there a reason for having before and after? Here there is no processing, except a `tell` between them.

Reply:
Looks like the before/after is used for tracing here.

Reply:
ok, thanks